pkg/handlers/license.go

```diff
@@ -334,7 +334,16 @@ func (h *Handler) UploadNewLicense(w http.ResponseWriter, r *http.Request) {
 	desiredAppName := strings.Replace(verifiedLicense.Spec.AppSlug, "-", " ", 0)
 	upstreamURI := fmt.Sprintf("replicated://%s", verifiedLicense.Spec.AppSlug)
 
-	a, err := store.GetStore().CreateApp(desiredAppName, upstreamURI, licenseString, verifiedLicense.Spec.IsAirgapSupported, installationParams.SkipImagePush, installationParams.RegistryIsReadOnly)
+	// verify that requested channel slug exists in the license
+	matchedChannelID, err := kotsutil.FindChannelIDInLicense(installationParams.RequestedChannelSlug, verifiedLicense)
+	if err != nil {
+		logger.Error(err)
+		uploadLicenseResponse.Error = "Your current license does not grant access to the channel you requested. Please generate a support bundle and contact support for assistance."
```

Comment:
should this error just be returned by `FindChannelIDInLicense` instead?

Reply:
I'm not sure, originally i was just returning the terser error that FindChannelIDInLicense returned but Brian word smith'd this for when this bubbles up in the UI. If we're cool with this more verbose one pop'ing up in other places (logs, cli, etc) i can definitely return it directly.